src/test/java/org/example/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,229 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.example.bookstore.dto.book.BookDto;
+import org.example.bookstore.dto.book.CreateBookRequestDto;
+import org.example.bookstore.exception.EntityNotFoundException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+            .webAppContextSetup(applicationContext)
+            .apply(springSecurity())
+            .build();
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("""
+        Create a new book with valid RequestDto is successful
+            """)
+    @Sql(scripts = "classpath:database/categories/insert-two-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+        "classpath:database/bookscategories/delete-from-books-categories.sql",
+        "classpath:database/books/delete-from-books.sql",
+        "classpath:database/categories/delete-from-categories.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void createBook_validRequestDto_successful() throws Exception {
+        CreateBookRequestDto requestDto = createBookRequestDto(1L);
+        BookDto expected = createBookDto();
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+        MvcResult result = mockMvc.perform(
+                post("/books")
+                .content(jsonRequest)
+                .contentType(MediaType.APPLICATION_JSON)
+            )
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto.class
+        );
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
```

Comment:
for other places also
```suggestion
        assertTrue(EqualsBuilder.reflectionEquals(expected, actual, "id"));
```

Reply:
not fixed 
try to test this `EqualsBuilder.reflectionEquals(expected, actual, "id"));` with valid and not valid data